### PR TITLE
Initialize CLI interface for python-zulip-api.

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

File: zulip/zulip/cli.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    log_exit(response)
+
+
+# Streams API
+
+
+@cli.command()
+def get_subscriptions() -> None:
+    """Get all streams that the user is subscribed to."""
+    # TODO python-zulip-api deviates from zulip.yaml here
```
Comment (timabbott) [inline zulip/zulip/cli.py:189]: I think python-zulip-api is wrong, and we should migrate it.  

Can you open an issue or quick PR?

**Comment by timabbott** [comment]
This is cool!  I think it'd really helpful to have a README (that perhaps we can then port into the /api documentation) with a handful of examples.  (I imagine it mostly self-documents with `--help`?)

File: zulip/zulip/cli.py (hunk 1 above)
Comment (rht) [reply zulip/zulip/cli.py:189]: This comment is obsolete. I already made the PR in #665. Already merged.

**Comment by rht** [comment]
@timabbott rebased, and used `get_subscriptions` instead of `list_subscriptions`.

**Comment by timabbott** [comment]
Merged, thanks @rht!  We'll want to do a bit more playing with this and then add it to the documentation.

@PIG208 @LoopThrough-i-j @showell are any of you up for spending some time playing with this, perhaps live-blogging the experience in #integrations?

File: zulip/zulip/cli.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+import logging
+import sys
+
+from typing import Any, Dict, List
+
+import zulip
+import click
```
Comment (andersk) [inline zulip/zulip/cli.py:8]: This `click` requirement is not declared in `setup.py`.

File: zulip/zulip/cli.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+log = logging.getLogger("zulip-cli")
+
+client = zulip.Client(config_file="~/zuliprc")
```
Comment (andersk) [inline zulip/zulip/cli.py:13]: This seems like a poor location for a config file, especially since it can’t be overridden.

Comment (rht) [reply zulip/zulip/cli.py:13]: Not sure the best way to handle this. There is already https://github.com/zulip/python-zulip-api/blob/717a549f4b097b7d08470bb860f0346c30d4ecd8/zulip_bots/zulip_bots/run.py#L55, but it's buried down inside zulip_bots. Should we move it to zulip library instead?

File: zulip/zulip/cli.py (hunk 2 above)
Comment (rht) [reply zulip/zulip/cli.py:8]: #694

File: zulip/zulip/cli.py (hunk 3 above)
Comment (timabbott) [reply zulip/zulip/cli.py:13]: I think we should just add the argparse logic from the core library for how to find a zuliprc file.

**Pull request merged**